src/execution/__tests__/lists-test.js

```diff
@@ -103,6 +104,30 @@ describe('Execute: Accepts async iterables as list value', () => {
       ],
     });
   });
+  it('Does not leak memory', async () => {
+    const mem = [];
+    async function* listField() {
+      for (let index = 0; index < 1000001; index++) {
+        if (index % 100000 === 0) {
+          mem.push([index, process.memoryUsage().heapUsed]);
+        }
+        yield index;
+      }
+    }
+    const memoryAtStart = process.memoryUsage().heapUsed;
+    await complete({ listField });
+    const memoryAtEnd = process.memoryUsage().heapUsed;
+
+    const results = regression.linear(mem);
+
+    assert(
+      results.r2 < 0.75 || results.equation[0] < 50,
+      'Expected non-linear growth of memory use. ' +
+        `Saw: ${mem.map((pt) => mb(pt[1])).join('->')}. Total Growth: ${mb(
+          memoryAtEnd - memoryAtStart,
+        )}`,
```

Comment:
@IvanGoncharov I adapted this test from [iterall](https://github.com/leebyron/iterall/blob/master/test.js#L1485-L1499). However, it will always fail because of the memory used to hold the GraphQL result in `completedResults`. You mentioned you had some ideas how to test this, do you think we can tweak the parameters or did you have something else entirely in mind?